src/TrashLib/Config/Secrets/SecretsProvider.cs

```diff
@@ -0,0 +1,34 @@
+using TrashLib.Startup;
+using YamlDotNet.Serialization;
+
+namespace TrashLib.Config.Secrets;
+
+public class SecretsProvider : ISecretsProvider
+{
+    public IDictionary<string, dynamic> Secrets
+    {
+        get => _secrets.Value;
+    }
+
+    private readonly IAppPaths _paths;
+    private readonly Lazy<Dictionary<string, dynamic>> _secrets;
+
+    public SecretsProvider(IAppPaths paths)
+    {
+        _paths = paths;
+        _secrets = new Lazy<Dictionary<string, dynamic>>(() => LoadSecretsFile());
+    }
+
+    private Dictionary<string, dynamic> LoadSecretsFile()
+    {
+        if (_paths.SecretsPath.Exists)
+        {
+            using var stream = _paths.SecretsPath.OpenText();
+            var deserializer = new DeserializerBuilder().Build();
```

Comment:
I think I got a little confused here. I think the way you have it might actually be perfectly fine. Because there's no "fixed schema" for the secrets YAML. It's really up to the user to specify what style to use. They could do `SomeCrazyKey: value`, or `some_crazy_key: value`, or `SOMECRAZYKEY: value`. As long as the key & its usage match, should be no issue I think?

Do you think it would make sense to have unit tests that prove out the different naming styles? It may not matter. In any case, if you feel comfortable with it, I think you can resolve this one with no change.

Reply:
Heh -- this is the #1 thing I expected to discuss during this review. 

I added the "bool" for the reason you pointed out: I wanted to use the YamlSerializerFactory but it can't require the secrets file to load the secrets file. However, once I added the "bool" then I ended up with a circular dependency in AutoFac container resolution.  I'll remove the bool and, if you're OK with it, I would like to keep the basic deserializer the way it is. Alternatively I would create an ISecretDeserializerFactory but I'm honestly not sure it's worth it. I defer to you. 

Regarding the naming, you are correct -- it's arbitrary.  I'll add a couple test cases but I expect people to be able to specify whatever they like and I'm just loading a dictionary.

Reply:
I am 100% OK with you keeping the basic deserializer as it is. No need to overcomplicate it. I apologize for my confusion :)